# Final results from regression models

## Accesing results from the three models

In [3]:
from pandas import DataFrame
from numpy import around

In [4]:
# Scores for linear regrresion
%store -r MS2_E_LR
%store -r MA_E_LR
%store -r R2_LR

# Scores for Decision Tree regression
%store -r MS2_E_DT
%store -r MA_E_DT
%store -r R2_DT

# Scores for Random Forest regression
%store -r MS2_E_RF
%store -r MA_E_RF
%store -r R2_RF

Scores_Models = dict(MS2=[MS2_E_LR, MS2_E_DT, MS2_E_RF],
                     MA=[MA_E_LR, MA_E_DT, MA_E_RF],
                     R2=[R2_LR, R2_DT, R2_RF])

Scores_Models = around(DataFrame(Scores_Models, index=["Linear", "Decision Tree", "Random Forest"]), 4)
Scores_Models

MS2      MA      R2
Linear         0.0460  0.0347  0.5723
Decision Tree  0.0479  0.0345  0.5373
Random Forest  0.0428  0.0317  0.6395

The first relevant observation for the three models is the "intermedium" values that we got with each one since the minimum and maximum $R^{2}$ are 0.537 and 0.639, respectively. The proposed cause for this to happen is that we have intrinsic bias in the data which is really hard to identify visually and even numerically (The outlier dropping with Mahalanobis distance shows this) and the same reason can be justified by considering the cultural context of the time and places in which the data was gather.

Also the same data doesn't have a natural origin as a scientific experiment or even a technological device which one would expect to be "precise" but comes from artificial paramaters to study human populations by other humans.

Furthermore, the three models can be described as followed:
- The linear model has a very good prediction of the violent crimes per population besides the simplicity of it with a $R^{2}$ of 0.572
- The Decision Tree model falls in the category of worst predictor with a $R^{2}$ of 0.537, the reason it performs worse than the linear model could be associated with the way decision trees work
- The Random Forest, even thouh it relies on decision trees, has the best prediction with a $R^{2}$ of 0.639 which highlights the strenght of this model